File: Easy/387. First Unique Character in a String/solution.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> hashMap(26, 0);
        
        for (char ch : s){
            hashMap[ch - 'a']++;
        }
        
        for (int i = 0; i < s.size(); i++){
            if (hashMap[s[i] - 'a'] == 1){
                return i;
            }
        }
        
        return -1;
    }
};

File: Easy/387. First Unique Character in a String/solution.cs
public class Solution {
    public int FirstUniqChar(string s) {
        Dictionary<char, int> myDictionary = new Dictionary<char, int>();
        foreach(char ch in s){
            if (myDictionary.ContainsKey(ch)){
                myDictionary[ch]++;
            }
            else{
                myDictionary.Add(ch, 1);
            }
        }

        for (int i = 0; i < s.Length; i++){
            if (myDictionary[s[i]] == 1)   return i;
        }
        return -1;
    }
}